cmsis/TARGET_CORTEX_A/cmsis_iccarm.h

```diff
@@ -581,6 +603,13 @@ void __FPU_Enable(void)
 }
 
 
+__STATIC_INLINE
+int __disable_irq_iar() {
```

Comment:
Its not the intention of CMSIS to incorporate device and/or compiler specific functions. All functions (API) shall rather be implemented for all compilers in this case. It seems you are looking for a extended version of __disable_irq() that returns the value of CPSR before modification.

Reply:
I reverted this function by cherry-picking from CMSIS repo, implemented disable_irq() for IAR by utilizing mbed critical function.